ttnn/cpp/ttnn/tensor/layout/tensor_layout.cpp

```diff
@@ -44,15 +44,40 @@ Alignment legacyShapeToAlignment(const ttnn::Shape& shape) {
 }
 
 TensorLayout::TensorLayout(DataType dtype, const PageConfig& page_config, const MemoryConfig& memory_config)
-    : TensorLayout(dtype, page_config, memory_config, {}) {
+    : TensorLayout(dtype, page_config, memory_config, std::nullopt, {}) {
+}
+
+TensorLayout::TensorLayout(DataType dtype, const PageConfig& page_config, const MemoryConfig& memory_config, const std::optional<Size>& shard_shape, const Alignment& shard_alignment)
+    : dtype_(dtype),
+      page_config_(page_config),
+      memory_config_(memory_config),
+      alignment_(shard_alignment) {
+
+    // TODO: If we remove alignment, we should lower this into validate_alignment
+    TT_FATAL(shard_alignment.size() <= 2, "Shard alignment {} exceeds maximum rank of 2", shard_alignment);
```

Comment:
Let's change the typo of this to have exactly two numbers. Possibly with a constructors from 0/1/2 numbers